src/main/java/org/example/controller/BookController.java

```diff
@@ -0,0 +1,45 @@
+package org.example.controller;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.example.dto.BookDto;
+import org.example.dto.CreateBookRequestDto;
+import org.example.dto.mapper.BookMapper;
+import org.example.exception.EntityNotFoundException;
+import org.example.model.Book;
+import org.example.service.BookService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/books")
+@RequiredArgsConstructor
+public class BookController {
+
+    private final BookService bookService;
+    private final BookMapper bookMapper;
+
+    @GetMapping
+    public List<BookDto> getAll() {
+        List<Book> books = bookService.findAll();
```

Comment:
yes, you can better write like this. I guess no need to write which line of code I have to remove and what to insert, so that I have only to copypaste, like in the comment above. But they what I have to do generally - would be better